advanced_alchemy/base.py

```diff
@@ -226,6 +219,17 @@ def to_dict(self, exclude: set[str] | None = None) -> dict[str, Any]:
         }
 
 
+class CommonTableAttributes(BasicAttributes):
+    """Common attributes for SQLALchemy tables."""
+
+    # noinspection PyMethodParameters
+    @declared_attr.directive
+    def __tablename__(cls) -> str:
+        """Infer table name from class name."""
+        regexp = re.compile("((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))")
```

Comment:
not sure if this is a nit or not, but as per [docs](https://docs.python.org/3/library/re.html#re.compile)
> using [re.compile()](https://docs.python.org/3/library/re.html#re.compile) and saving the resulting regular expression object for reuse is more efficient when the expression will be used several times in a single program.

maybe promote this to `ClassVar`?

Reply:
well, if we are going to modify this.  Is there a way that can type this so that it can be overridden on a per model basis as a string value without a mypy warning?